### PR TITLE
User interface milestone 2

### PR DESCRIPTION
**Pull request by HillaryOkello**
In this milestone, we created the user interface for the tic-tac-toe game.

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team 😄,

**Your project shows great initiative 👏. However, a few things need to be implemented before it can be approved.**

1). **Basic Hint.**
image

2). **The main goal of this milestone is to draw a design of the game. So you need to show the following,**

-  How the game will tell which player's turn it is. 

- How the game will ask the player to select the available moves.

- How the game board will look like after a move has been played.

- Describe an invalid move scenario and show how the game will inform the player that they have made an invalid move.

- You have shown how the game will inform of a win, but you haven't described a winning move scenario. You can use a game board for this.

- You have shown how the game will inform of a draw but you haven't described a  game draw scenario. You can use a game board for this.

3). **You need to define a loop to describe that you are aware that some repetitive tasks are gonna be placed in one. Something like this,**

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw 
    game_on = false
  end
end
```

## Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

File: bin/main.rb [hunk 1]
```diff
@@ -1 +1,63 @@
...
+    puts 'Valid move'
+  end
+
+  winner = true
+  # player tokens are alligned in the entire row or column
+  if winner
+    puts 'You won'
+    game_on = false
+  else
+    puts 'You Drew'
```
Comment (mcrd25) [inline bin/main.rb]: Prefer message: It's a Draw

File: bin/main.rb [hunk 2]
```diff
@@ -1 +1,63 @@
...
+  puts " #{board[3]} | #{board[4]} | #{board[5]} "
+  puts '-----------'
+  puts " #{board[6]} | #{board[7]} | #{board[8]} "
+end
+display_board(board)
+
+game_on = true
+
+puts 'Please enter your token (X or O)'
+token = gets.chomp!
```
Comment (mcrd25) [inline bin/main.rb:32]: [OPTIONAL]
How do we know which player this is for?

File: bin/main.rb [hunk 3]
```diff
@@ -1 +1,63 @@
-puts 'Hello World!'
+puts "Welcome to Pascal and Hillary's tic-tac-toe game"
```
Comment (mcrd25) [inline bin/main.rb:2]: /you are missing the line to make this file executable.
`#!/usr/bin/env ruby` needs to be added and the right permissions need to be added. Please see this [screenshot](requirements_screenshots)

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 

Hi Team! 👋 

A few changes required but you've done a great job mocking the flow of the game!

### Required changes
- Your game asks the player to select from available moves
- Check my review comments

***Please makes changes and submit another review request.
If you have any questions, let me know here or email (preferably) or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

If you liked my review, consider upskilling me on linkedin. And don't be afraid to connect. [My LinkedIn](https://www.linkedin.com/in/mayadouglas/)

*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work, team. 💻 👀 ✨ 

- Good job implementing the changes requested previously.

## However, there are some changes required 👀 🧰 🔨 🔧 

- Let's improve the user interface by implementing a game loop. Use all the methods you've developed so far, but now add a game loop structure such as this, to have a basic flow for the players to follow:

```
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Great job guys, well done with the project.

now you can move to the next milestone, this milestone is approved.

## Happy Coding :tophat:

**Pull request merged**